api/api.go

```diff
@@ -132,8 +135,14 @@ func New(opts Options) (*API, error) {
 		return nil, err
 	}
 
-	// TODO(beorn7): For now, this hardcodes the method="get" label. Other
-	// methods should get the same instrumentation.
+	latency := prometheus.NewHistogramVec(
```

Comment:
Thinking about this a bit more. What if we pass the `alertmanager_http_request_duration_seconds` metric var from `main.go` to `api.New()`? We'd need to add `code` to the metric, but I think it would make more sense to have a single metric since it's all on the same HTTP server/mux.